server/src/test/java/org/elasticsearch/persistent/PersistentTasksNodeServiceTests.java

```diff
@@ -231,7 +232,9 @@ public void testParamsStatusAndNodeTaskAreDelegated() throws Exception {
     public void testTaskCancellation() {
         AtomicLong capturedTaskId = new AtomicLong();
         AtomicReference<ActionListener<CancelTasksResponse>> capturedListener = new AtomicReference<>();
-        PersistentTasksService persistentTasksService = new PersistentTasksService(null, null, null) {
```

Comment:
`null` didn't work here because `PersistentTasksService` uses the `OriginSettingsClient` which extends `FilterClient` which fetches things from the client that you pass it.